pandas/tests/reshape/concat/test_categorical.py

```diff
@@ -202,3 +204,23 @@ def test_categorical_concat_gh7864(self):
 
         dfa = df1.append(df2)
         tm.assert_index_equal(df["grade"].cat.categories, dfa["grade"].cat.categories)
+
+    def test_categorical_datetime_concat(self):
+        # GH 39443
+        df1 = DataFrame(
+            {"x": Series(datetime(2021, 1, 1), index=[0], dtype="category")}
+        )
+        df2 = DataFrame(
+            {"x": Series(datetime(2021, 1, 2), index=[1], dtype="category")}
+        )
+
+        result = pd.concat([df1, df2])
+        expected = DataFrame(
+            {
+                "x": Series(
+                    [datetime(2021, 1, 1), datetime(2021, 1, 2)], dtype="category"
```

Comment:
Referencing some of your exploration on the related issue, I think not returning a categorical on this concatenation is expected since the inputs don't have same categorical dtype (eg different categories). 

The same behavior is seen if you just do something like replace the input to `df1` and `df2` with 1 and 2, respectively (gives int dtype result, not categorical).

Reply:
That makes a lot of sense. I've rebased to my original commit removing the categorical dtype in the `expected` dataframe. However, I think the tests are going to fail again with seemingly contradictory messages.

Reply:
on 1.1.5 the return type is datetime64[ns] and this is also the case on 1.2.4 and master. So testing that this matches the 1.1.5 behavior is fine.